src/utils/axiosInterceptors.ts

```diff
@@ -6,36 +6,34 @@ const EXPIRY_HEADER_KEY = 'Cache_Expires';
 const EXPIRES_IN_SECONDS = 60 * 30;
 const DELAY = 3000;
 const RETRIES = 5;
-export interface RequestConfig extends AxiosRequestConfig {
-  useCache?: boolean;
-  retries?: number;
+
+// we are extending axios' AxiosRequestConfig:
+// https://stackoverflow.com/questions/58777924/axios-typescript-customize-axiosrequestconfig
+declare module 'axios' {
+  export interface AxiosRequestConfig {
+    useCache?: boolean;
+    retries?: number;
+  }
 }
 
 export const registerAxiosCacheRetryInterceptors = (): any => {
   findAndDeleteExpiredCachedItems();
   axios.interceptors.request.use(fetchCachedResponse, error => Promise.reject(error));
-  axios.interceptors.response.use(setCacheResponse, retryRequests);
+  axios.interceptors.response.use(saveCacheResponse, error => retryRequests(error));
 };
 
 const fetchCachedResponse = async (request: any): Promise<any> => {
   if (!(request && request.useCache)) {
     return request;
   }
-
-  let cacheKey;
-  switch (request.method) {
-    // post requests are not supported, so we mimic a get request, by formatting the body/params to sha256, and constructing a key/url
-    // idea taken from https://blog.cloudflare.com/introducing-the-workers-cache-api-giving-you-control-over-how-your-content-is-cached/
-    case 'post':
-      const body = JSON.stringify(request.data);
-      const hash = await sha256(body);
-      cacheKey = generateCacheKey(request.url, hash);
-      break;
-    case 'get':
-      cacheKey = generateCacheKey(request.url, stringify(request.params));
-      break;
-    default:
-      return request;
+  // do not perform caching if Cache API is not supported:
+  if (typeof window === 'undefined' || !window.caches) {
+    return request;
+  }
+  // resource not cacheable? It couldn't have been saved to cache:
```

Comment:
Maybe move this comment after `const cacheKey = await generateCacheKey(request);` so it wouldn't look like `cacheKey ` is used just for checking if the item is *not* cached.

Reply:
Thanks, good point! Fixed.

Reply:
Great. Resolving.